src/main/java/com/mongodb/spark/sql/connector/read/MongoContinuousStream.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.mongodb.spark.sql.connector.read;
+
+import static com.mongodb.spark.sql.connector.read.ResumeTokenOffset.INITIAL_RESUME_TOKEN_OFFSET;
+import static java.lang.String.format;
+import static java.util.Collections.emptyList;
+
+import org.apache.spark.sql.connector.read.InputPartition;
+import org.apache.spark.sql.connector.read.streaming.ContinuousPartitionReaderFactory;
+import org.apache.spark.sql.connector.read.streaming.ContinuousStream;
+import org.apache.spark.sql.connector.read.streaming.Offset;
+import org.apache.spark.sql.connector.read.streaming.PartitionOffset;
+import org.apache.spark.sql.types.StructType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.mongodb.spark.sql.connector.assertions.Assertions;
+import com.mongodb.spark.sql.connector.config.ReadConfig;
+import com.mongodb.spark.sql.connector.schema.BsonDocumentToRowConverter;
+
+/** MongoContinuousStream defines how to read a stream of data from MongoDB. */
+public class MongoContinuousStream implements ContinuousStream {
+  private static final Logger LOGGER = LoggerFactory.getLogger(MongoContinuousStream.class);
+  private final BsonDocumentToRowConverter bsonDocumentToRowConverter;
+  private final ReadConfig readConfig;
+  private int partitionId = -1;
+
+  /**
+   * Construct a new instance
+   *
+   * @param schema the schema for the data
+   * @param readConfig the read configuration
+   */
+  public MongoContinuousStream(final StructType schema, final ReadConfig readConfig) {
+    Assertions.validateConfig(
+        schema,
+        (s) -> !s.isEmpty(),
+        () -> "Mongo Continuous streams require a schema to be defined");
+    this.bsonDocumentToRowConverter = new BsonDocumentToRowConverter(schema);
+    this.readConfig = readConfig;
+  }
+
+  @Override
+  public InputPartition[] planInputPartitions(final Offset start) {
+    this.partitionId += 1;
```

Comment:
Given that `MongoContinuousStream.planInputPartitions` always produces exactly one partition and is called not more than once per instance of `MongoStreamPartitionReader`, because `MongoContinuousStream.needsReconfiguration` always returns `false`, why do we have the `partitionId` field? It appears that `planInputPartitions` should just pass 0 as the partition ID when it creates an instance of `MongoInputPartition`.